exchangesync/node.go

```diff
@@ -106,6 +106,17 @@ func NodeInitalSetup(db *bolt.DB, getDevice exchange.DeviceHandler, patchDevice
 				return fmt.Errorf("Unable to update the Exchange with correct node version. %v", err)
 			}
 		}
+
+		cluster_namespace := ""
+		if pDevice.Config.State == persistence.CONFIGSTATE_CONFIGURED && pDevice.GetNodeType() == persistence.DEVICE_TYPE_CLUSTER {
+			cluster_namespace = cutil.GetClusterNamespace()
+			if exchNode.ClusterNamespace != cluster_namespace {
+				// update exchange with correct clusterNamespace
+				if err = patchDevice(fmt.Sprintf("%v/%v", pDevice.Org, pDevice.Id), pDevice.Token, &exchange.PatchDeviceRequest{ClusterNamespace: &cluster_namespace}); err != nil {
+					return fmt.Errorf("Unable to update the Exchange with correct node cluster namespace. %v", err)
+				}
+			}
+		}
```

Comment:
We should also clear the cluster namespace for the exchange node if node type is device.